src/main/java/org/opensearch/knn/jni/JNIService.java

```diff
@@ -13,16 +13,99 @@
 
 import org.apache.commons.lang.ArrayUtils;
 import org.opensearch.common.Nullable;
+import org.opensearch.knn.common.KNNConstants;
 import org.opensearch.knn.index.IndexUtil;
 import org.opensearch.knn.index.query.KNNQueryResult;
 import org.opensearch.knn.index.util.KNNEngine;
 
+import java.util.Locale;
 import java.util.Map;
 
 /**
  * Service to distribute requests to the proper engine jni service
  */
 public class JNIService {
+    /**
+     * Initialize an index for the native library. Takes in numDocs to
+     * allocate the correct amount of memory.
+     *
+     * @param numDocs number of documents to be added
+     * @param dim dimension of the vector to be indexed
+     * @param parameters parameters to build index
+     * @param knnEngine knn engine
+     * @return address of the index in memory
+     */
+    public static long initIndexFromScratch(long numDocs, int dim, Map<String, Object> parameters, KNNEngine knnEngine) {
```

Comment:
Since we are creating this functions from Scratch , Let's wrap these parameters in class.

Reply:
Resolved as this will be a change when JNI refactoring is done.